File: solution-python/026_remove_duplicates_from_sorted_array.py
"""
	026. Remove Duplicates from Sorted Array
	@name: li jin
	@date: Mar 03, 2016
	@link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/
	@time: 92 ms
"""

class Solution(object):
    def removeDuplicates(self, nums):
        if len(nums) < 2: return len(nums)
        pos = 1
        for i in xrange(1, len(nums)):
            if nums[i] != nums[i-1]:
                nums[pos] = nums[i]
                pos += 1
        del nums[pos:]
        return pos

File: solution-cpp/026_remove_duplicates_from_sorted_array.cpp
/*
	026. Remove Duplicates from Sorted Array
	@name: li jin
	@date: Mar 03, 2016
	@link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/
	@time: 32 ms
*/

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 2) return nums.size();
        int res = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] != nums[i-1]) nums[res++] = nums[i];
        }
        nums.erase(nums.begin() + res, nums.end());
        return res;
    }
};
